src/cmd/product.js

```diff
@@ -20,22 +23,86 @@ module.exports = class ProductCommand {
 		spinnerMixin(this);
 	}
 
+	addDevice({ params: { product, device } }){
+		const identifiers = [device];
+		const msg = `Adding device ${device} to product ${product}`;
+		const upload = uploadProductDevices(product, identifiers);
+		return this.showBusySpinnerUntilResolved(msg, upload)
+			.then(result => this.showDeviceAddResult(result));
+	}
+
+	addDevices({ file, params: { product, device } }){
+		if (!device && !file){
+			throw usageError(
+				'`device` parameter or `--file` option is required'
+			);
+		}
+
+		if (device){
+			return this.addDevice({ params: { product, device } });
+		}
+
+		const msg = `Adding devices in ${file} to product ${product}`;
+		const upload = readDeviceListFile(file).then((identifiers) => uploadProductDevices(product, identifiers));
+		return this.showBusySpinnerUntilResolved(msg, upload)
+			.then(result => this.showDeviceAddResult(result));
+	}
+
+	showDeviceAddResult({ product, identifiers, status: { invalidDeviceIds = [], nonmemberDeviceIds = [] } } = {}){
```

Comment:
one of these days, seeing array/object defaults in JS wont make me panic like python taught me to